ts/features/pn/aar/store/reducers/index.ts

```diff
@@ -0,0 +1,145 @@
+import { ThirdPartyMessage } from "../../../../../../definitions/pn/ThirdPartyMessage";
+import { Action } from "../../../../../store/actions/types";
+import { GlobalState } from "../../../../../store/reducers/types";
+import { setAarFlowState, terminateAarFlow } from "../actions";
+
+export const sendAARFlowStates = {
+  none: "none",
+  displayingAARToS: "displayingAARToS",
+  fetchingQRData: "fetchingQRData",
+  fetchingNotificationData: "fetchingNotificationData",
+  displayingNotificationData: "displayingNotificationData",
+  notAddresseeFinal: "notAddresseeFinal",
+  ko: "ko"
+} as const;
+
+export const validAARStatusTransitions = new Map<
+  AARFlowState["type"],
+  Set<AARFlowState["type"]>
+>([
+  [sendAARFlowStates.none, new Set([sendAARFlowStates.displayingAARToS])],
+  [
+    sendAARFlowStates.displayingAARToS,
+    new Set([sendAARFlowStates.fetchingQRData])
+  ],
+  [
+    sendAARFlowStates.fetchingQRData,
+    new Set([sendAARFlowStates.fetchingNotificationData, sendAARFlowStates.ko])
+  ],
+  [
+    sendAARFlowStates.fetchingNotificationData,
+    new Set([
+      sendAARFlowStates.displayingNotificationData,
+      sendAARFlowStates.notAddresseeFinal,
+      sendAARFlowStates.ko
+    ])
+  ],
+  [
+    sendAARFlowStates.displayingNotificationData,
+    new Set([sendAARFlowStates.ko])
+  ],
+  [sendAARFlowStates.notAddresseeFinal, new Set([sendAARFlowStates.ko])],
+  [
+    sendAARFlowStates.ko,
+    new Set([
+      sendAARFlowStates.fetchingQRData,
+      sendAARFlowStates.fetchingNotificationData,
+      sendAARFlowStates.displayingNotificationData,
+      sendAARFlowStates.none
+    ])
+  ]
+]);
+
+type FlowStates = typeof sendAARFlowStates;
+
+type NotInitialized = {
+  type: FlowStates["none"];
+};
+
+type DisplayingTos = {
+  type: FlowStates["displayingAARToS"];
+  qrCode: string;
+};
+
+type FetchQR = {
+  type: FlowStates["fetchingQRData"];
+  qrCode: string;
+};
+
+type FetchNotification = {
+  type: FlowStates["fetchingNotificationData"];
+  iun: string;
+  fullNameDestinatario: string;
+  mandateId?: string;
+};
+
+type DisplayingNotification = {
+  type: FlowStates["displayingNotificationData"];
+  fullNameDestinatario: string;
+  notification: ThirdPartyMessage;
+  mandateId?: string;
+};
+
+type FinalNotAddressee = {
+  type: FlowStates["notAddresseeFinal"];
+  fullNameDestinatario: string;
+  qrCode: string;
+  iun: string;
+};
+
+type ErrorState = {
+  type: FlowStates["ko"];
+  errorKind?: string;
+  previousState: AARFlowState;
+};
+
+export type AARFlowState =
+  | NotInitialized
+  | DisplayingTos
+  | FetchQR
+  | FetchNotification
+  | DisplayingNotification
+  | FinalNotAddressee
+  | ErrorState;
+
+export const INITIAL_AAR_FLOW_STATE: AARFlowState = {
+  type: "none"
+};
+
+export const isValidAARStateTransition = (
+  currentState: AARFlowState,
+  nextState: AARFlowState
+): boolean => {
+  const currentType = currentState.type;
+  const nextType = nextState.type;
```

Comment:
```suggestion
  currentState: FlowStates[keyof FlowStates],
  nextState: FlowStates[keyof FlowStates]
): boolean => {
  const currentType = currentState;
  const nextType = nextState;
```

since you don't need anything from the rest of the state, I believe that passing only the type would make this cleaner -- what do you think?

Reply:
Yes you are right, I inserted your edit